### PR TITLE
CPDP-1423: Disposal date error message fix

### PR DESCRIPTION
**Pull request by rambabu-posa**

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/PropertyDetailsController.scala [hunk 1]
```diff
@@ -726,6 +730,11 @@ class PropertyDetailsController @Inject() (
         _ => routes.PropertyDetailsController.checkYourAnswers()
       )
 
+  private def getCompletionDate(answers: MultipleDisposalsTriageAnswers): CompletionDate =
+    answers
+      .fold(_.completionDate, c => Some(c.completionDate))
+      .getOrElse(CompletionDate(LocalDateUtils.today()))
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/PropertyDetailsController.scala]: we shouldn't default to today - we should redirect to the task list if we can't find a completion date in the triage answers

Comment (AndyHWChung) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/PropertyDetailsController.scala]: in your pattern match for the tax year you could possible add in a check for the completion date as well at the same time:
```scala
 m.triageAnswers.fold(i => i.taxYear -> i.completionDate, c => Some(c.taxYear) -> Some(c.completionDate))) match {
  case (Some(taxYear), Some(completionDate)) => ...
   case _ => Redirect(taskList)
```

Comment (rambabu-posa) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/PropertyDetailsController.scala]: Yes I liked this approach

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/PropertyDetailsController.scala [hunk 2]
```diff
@@ -384,19 +384,21 @@ class PropertyDetailsController @Inject() (
         r.draftReturn match {
           case _: SingleDisposalDraftReturn => Redirect(routes.PropertyDetailsController.checkYourAnswers())
           case m: MultipleDisposalsDraftReturn =>
-            m.triageAnswers.fold(_.taxYear, c => Some(c.taxYear)) match {
-              case Some(taxYear) =>
+            m.triageAnswers.fold(
+              i => i.taxYear       -> i.completionDate,
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/PropertyDetailsController.scala]: I think we're just missing unit tests to say what happens when the completion date isn't there now

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**